spec/support/model_helpers.rb

```diff
@@ -0,0 +1,26 @@
+# use like this:
+# it { is_expected.to have_validation :valid_date_presence,
+#                                     :on => :start_date,
+#                                     :options => [:allow_blank, :allow_nil]
+#                                     }
+RSpec::Matchers.define :have_validation do |validator, params|
+  validator_class = eval(validator.to_s.camelize + "Validator")
+
+  match do |model|
+    my_validators = model.class.validators.select { |v| v.class == validator_class and v.attributes.include? params[:on] }
+    my_validators = my_validators.select { |v| params[:options].all? { |po| v.options.keys.any? { |vo| vo.to_sym == po } } } if params[:options]
```

Comment:
I would really like to see this clarified. Here's my attempt, though I'm not sure how well this all will work in the context of an RSpec custom matcher.

Testing that a particular validator is only used once for a particular attribute is a good idea. Is this the best way to handle that?

The logic for options is as follows: if `params[:options]` are passed to the matcher, ensure that the validator in the model gets passed the same options, correct?
```ruby
match do |model|
  @validator = model.class.validators.select { |v| (v.class == validator_class) && v.attributes.include? params[:on] }
  validator_included_once? && validator_receives_correct_options?
end

def validator_included_once?
  @validator.size == 1
end

def validator_receives_correct_options?
  return true unless params[:options]
  params[:options].all? { |o| @validator.first.options.has_key?(o) }
end
```